components/st-microelectronics/stm-api.stanza

```diff
@@ -130,132 +138,177 @@ defn match-pin? (pin-ref:Ref power:String) :
   if p-ref == power :
     true
 
-defn find-pins-list (obj:JITXObject my-pin:String) :
+; return list of pins that match name (exactly) / don't check for empty
+defn get-pin-list (obj:JITXObject pin-id:?) :
+  val pin-name = to-string(pin-id)
   val result = to-tuple $ for p in pins(obj) filter :
-    if substring?(to-string(ref(p)), my-pin) :
-      match-pin?(ref(p), my-pin)
+    if substring?(to-string(ref(p)), pin-name) :
+      match-pin?(ref(p), pin-name)
   result
 
-defn find-pins? (obj:JITXObject my-pin:String) :
-  val result = to-tuple $ for p in pins(obj) filter :
-    if substring?(to-string(ref(p)), my-pin) :
-      match-pin?(ref(p), my-pin)
+; find the pin exact match return a boolean
+defn find-pin? (obj:JITXObject my-pin:String|PwrPinNames|GndPinNames|ResetPinNames|BootPinNames|ReferencePinNames) :
+  val result = get-pin-list(obj, my-pin)
   not empty?(result)
 
-defn power-pins (stm:JITXObject vdd-pin:PowerPinNames) -> Tuple<JITXObject> :
-  val found-pins = find-pins-list(stm, to-string(vdd-pin))
+defn power-pins (stm:JITXObject vdd-pin:PwrPinNames) -> Tuple<JITXObject> :
+  val found-pins = get-pin-list(stm, vdd-pin)
   if empty?(found-pins) :
-    fatal("%_ is missing an identifiable power pin." % [ref(stm)])
+    fatal("%_ is missing an identifiable power pin (%_)." % [ref(stm), to-string(found-pins[0])])
```

Comment:
This will break --
`found-pins` is empty so its first item cannot be accessed.

Reply:
Ah, good catch